eclipse-collections-code-generator-maven-plugin/src/main/java/org/eclipse/collections/codegenerator/maven/GenerateMojo.java

```diff
@@ -63,27 +63,23 @@ public void execute() throws MojoExecutionException
         List<URL> urls = Arrays.asList(((URLClassLoader) GenerateMojo.class.getClassLoader()).getURLs());
 
         boolean[] error = new boolean[1];
-        EclipseCollectionsCodeGenerator.ErrorListener errorListener = new EclipseCollectionsCodeGenerator.ErrorListener()
-        {
-            public void error(String string)
-            {
-                GenerateMojo.this.getLog().error(string);
-                error[0] = true;
-            }
+        EclipseCollectionsCodeGenerator.ErrorListener errorListener = string -> {
+            this.getLog().error(string);
+            error[0] = true;
         };
-        EclipseCollectionsCodeGenerator gsCollectionsCodeGenerator =
+        EclipseCollectionsCodeGenerator codeGenerator =
                 new EclipseCollectionsCodeGenerator(this.templateDirectory, this.project.getBasedir(), urls, errorListener);
         if (!this.skipCodeGen)
         {
-            int numFilesWritten = gsCollectionsCodeGenerator.generateFiles();
+            int numFilesWritten = codeGenerator.generateFiles();
             this.getLog().info("Generated " + numFilesWritten + " files");
         }
         if (error[0])
         {
             throw new MojoExecutionException("Error(s) during code generation.");
         }
 
-        if (gsCollectionsCodeGenerator.isTest())
+        if (codeGenerator.isTest())
         {
             this.project.addTestCompileSourceRoot(EclipseCollectionsCodeGenerator.GENERATED_TEST_SOURCES_LOCATION);
         }
```

Comment:
And here's the place that would need to change so you don't have to manually set up resources directories. Something like:

```
        Resource resource = new Resource();
        resource.setDirectory(outputDirectory);
        this.project.addResource(resource);
```

This stuff is tricky, I can probably help with another PR on top.

Reply:
I originally tried something similar to this and couldn't get it to work. Please feel free to merge this PR and then submit another on top with the changes. I'll be happy to review and merge.

Reply:
@motlin Do you want to merge and apply changes in a separate PR?